### PR TITLE
Add an incrementValue to Storage adapters

### PR DESCRIPTION
**Pull request by regularjack**
We need to be able to perform an atomic increment

### CODE REVIEW

File: src/Flaps/Storage/DoctrineCacheAdapter.php [hunk 1]
```diff
@@ -42,6 +42,14 @@ public function setValue($key, $value)
         $this->cache->save($key, intval($value));
     }
 
+    public function incrementValue($key)
+    {
+        $value = $this->getValue($key) + 1;
```
Comment (mrnnn) [inline src/Flaps/Storage/DoctrineCacheAdapter.php]: Only one thought here: we are kind of suggesting that increment is atomic but it isn't here.

an idea would be to run an update query (update counters set amount = amount+1 where id = ? kind of query). Then, of course, getting the count becomes subject to race conditions. which is why i thought a notimplemented exception would be in place :-)

Comment (mrnnn) [inline src/Flaps/Storage/DoctrineCacheAdapter.php]: Or add a transaction of course

File: tests/Flaps/Mock/Storage.php [hunk 2]
```diff
@@ -22,6 +22,14 @@ public function setValue($key, $value)
         $this->values[$key] = $value;
     }
 
+    public function incrementValue($key)
+    {
+        $value = $this->getValue($key) + 1;
```
Comment (mrnnn) [inline tests/Flaps/Mock/Storage.php]: No issue here of course since mock.

File: src/Flaps/Storage/DoctrineCacheAdapter.php (hunk 1 above)
Comment (regularjack) [inline src/Flaps/Storage/DoctrineCacheAdapter.php]: AFAICT it's not possible to make it atomic so now throwing an exception

**Pull request merged**